app/src/org/commcare/preferences/DeveloperPreferences.java

```diff
@@ -72,45 +75,87 @@
+        prefKeyToAnalyticsEvent.put(MARKDOWN_ENABLED, GoogleAnalyticsFields.LABEL_MARKDOWN);
+        prefKeyToAnalyticsEvent.put(ALTERNATE_QUESTION_LAYOUT_ENABLED, GoogleAnalyticsFields.LABEL_IMAGE_ABOVE_TEXT);
+        prefKeyToAnalyticsEvent.put(HOME_REPORT_ENABLED, GoogleAnalyticsFields.LABEL_REPORT_BUTTON_ENABLED);
+        prefKeyToAnalyticsEvent.put(AUTO_PURGE_ENABLED, GoogleAnalyticsFields.LABEL_AUTO_PURGE);
+        prefKeyToAnalyticsEvent.put(LOAD_FORM_PAYLOAD_AS, GoogleAnalyticsFields.LABEL_LOAD_FORM_PAYLOAD_AS);
+        prefKeyToAnalyticsEvent.put(DETAIL_TAB_SWIPE_ACTION_ENABLED, GoogleAnalyticsFields.LABEL_DETAIL_TAB_SWIPE_ACTION);
+        prefKeyToAnalyticsEvent.put(PREFS_CUSTOM_RESTORE_DOC_LOCATION, GoogleAnalyticsFields.LABEL_CUSTOM_RESTORE);
     }
 
     /**
      * Spacer to distinguish between the saved navigation session and form entry session
      */
     private static final String NAV_AND_FORM_SESSION_SPACER = "@@@@@";
 
-    private static final Map<String, String> prefKeyToAnalyticsEvent = new HashMap<>();
     private Preference savedSessionEditTextPreference;
 
+    @NonNull
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        GoogleAnalyticsUtils.reportPrefActivityEntry(GoogleAnalyticsFields.CATEGORY_DEV_PREFS);
-        populatePrefKeyToEventLabelMapping();
-        initAllPrefs();
+    protected String getTitle() {
+        return "Developer Options";
     }
 
-    private void initAllPrefs() {
-        PreferenceManager prefMgr = getPreferenceManager();
-        prefMgr.setSharedPreferencesName((CommCareApplication.instance().getCurrentApp().getPreferencesFilename()));
-        addPreferencesFromResource(R.xml.preferences_developer);
-        setTitle("Developer Options");
+    @Nullable
+    @Override
+    protected Map<String, String> getPrefKeyAnalyticsEventMap() {
+        return prefKeyToAnalyticsEvent;
+    }
 
-        GoogleAnalyticsUtils.createPreferenceOnClickListeners(
-                prefMgr, prefKeyToAnalyticsEvent, GoogleAnalyticsFields.CATEGORY_DEV_PREFS);
+    @Override
+    protected void setupPrefClickListeners() {
+        createOnCustomRestoreOption();
+    }
 
+    @Nullable
+    @Override
+    protected Map<String, String> getPrefKeyTitleMap() {
+        return null;
+    }
+
+    @NonNull
+    @Override
+    protected int getPreferencesResource() {
+        return R.xml.preferences_developer;
+    }
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        super.onCreatePreferences(savedInstanceState, rootKey);
         savedSessionEditTextPreference = findPreference(EDIT_SAVE_SESSION);
+    }
+
+    @Override
+    protected boolean isAppLevelPreference() {
+        return true;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        hideOrShowDangerousSettings();
```

Comment:
Again would prefer onStart() for it as we are setting up what prefs to show here which should be done as soon as UI becomes visible.